tests/unit/test_metrics.py

```diff
@@ -79,3 +97,180 @@ def test_angle_relative_to_horizontal(
     assert angle_z_results.measurements.shape == (1,), (
         f"Expected anglez shape: {(1,)}," f"got({angle_z_results.measurements.shape})"
     )
+
+
+@pytest.mark.parametrize(
+    "nonwear_value_array, expected_result",
+    [
+        (
+            np.array([0, 1, 1, 0, 1, 1, 1]),
+            np.array([0, 1, 1, 0, 1, 1, 1]),
+        ),
+        (
+            np.array([0, 1, 1, 2, 1, 2, 1, 3]),
+            np.array([0, 1, 1, 2, 2, 2, 2, 3]),
+        ),
+    ],
+)
+def test_cleanup_isolated_ones_nonwear_value(
+    nonwear_value_array: np.ndarray, expected_result: np.ndarray
+) -> None:
+    """Test the cleanup isolated ones nonwear value function."""
+    test_result = metrics._cleanup_isolated_ones_nonwear_value(nonwear_value_array)
+
+    assert np.all(
+        test_result == expected_result
+    ), f"Expected {expected_result}, got: {test_result}"
+
+
+def test_group_acceleration_data_by_time() -> None:
+    """Test the group acceleration data by time function."""
+    dummy_date = datetime(2024, 5, 2)
+    dummy_datetime_list = [dummy_date + timedelta(seconds=i) for i in range(1001)]
+    test_time = pl.Series(dummy_datetime_list)
+    measurements = np.ones((1001, 3))
+    acceleration = models.Measurement(measurements=measurements, time=test_time)
+    window_length = int(10)
+    expected_time_length = math.ceil(len(test_time) / window_length)
+    expected_result_shape = acceleration.measurements.shape[1] + 1
+
+    result = metrics._group_acceleration_data_by_time(acceleration, window_length)
+
+    assert isinstance(result, pl.DataFrame)
+    assert result.shape[0] == expected_time_length
+    assert result.shape[1] == expected_result_shape
+
+
+@pytest.mark.parametrize(
+    "modifier, expected_result",
+    [
+        (1, 1),
+        (0, 0),
+    ],
+)
+def test_compute_nonwear_value_per_axis(
+    create_acceleration: pl.DataFrame, modifier: int, expected_result: int
+) -> None:
+    """Test the nonwear value per axis function."""
+    std_criteria = modifier
+    range_criteria = modifier
+    acceleration = create_acceleration.with_columns(pl.col("time").set_sorted())
+    acceleration_grouped = acceleration.group_by_dynamic(
+        index_column="time", every="5s"
+    ).agg([pl.all().exclude(["time"])])
+
+    test_resultx = metrics._compute_nonwear_value_per_axis(
+        acceleration_grouped["X"], std_criteria, range_criteria
+    )
+
+    assert (
+        test_resultx == expected_result
+    ), f"Expected {expected_result}, got: {test_resultx}"
+
+
+@pytest.mark.parametrize(
+    "modifier, expected_result",
+    [
+        (1, 3),
+        (1, 2),
+        (1, 1),
+        (0, 0),
+    ],
+)
+def test_compute_nonwear_value_array(
```

Comment:
If each of your parameterizations requires a separate if-statement, its time to break up your test; I now have to understand the complete if-statement and its relationship to your parameter to understand what the test is doing.